elastic/integrations
# [ti_eclecticiq][Enhancement] Enable SSL configuration

## Proposed commit message

### Enhancement: Enable SSL configuration

Motivation: some users use EIQ IC platform with invalid SSL certificates and don't intend to renew or add a valid certificate. Still want to use that instance at least for testing purpose and we need to enable them.

Implementation: added "SSL Configuration" yaml field to allow users to configure outgoing requests SSL certificate validation. This way they have full control over this part of the communication and they can turn off SSL validation by setting `verification_mode` to `none`.

### Bugfix: parsing incoming data fields that contain dots

Bug: conditions for dot expanders are bad and they prevent reading input data properly. This causes data loss.

Fix: the conditions are updated (fixed) to enable parsing incoming data fields that contain dots.

## Checklist


## Author's Checklist

- [ ] changelog.yml is updated according to standards

## How to test this PR locally

Scenario: SSL configuration field works as desired.
Given: updated integration is installed in an Elastic Stack environment
And: EIQ IC is running with on https with invalid certificate
And: integration is set up to use that IC instance with `verification_mode=none`
When: integration agent runs
Then: communication is successful.

Scenario: dot expanders are set up properly.
Given: updated integration is installed in an Elastic Stack environment
And: observables that are added to dataset to be downloaded contain populated fields that should be "dot expanded" (source.names, meta.tags, meta.ingest_time, meta.estimated_threat_start_time, meta.estimated_threat_end_time, meta.estimated_observed_time)
When: integration agent runs
Then: values from "dot expanded" fields are mapped into ECS fields (event.provider, tags, event.created, event.start, event.end, threat.indicator.first_seen).
